plugin/highlights.py

```diff
@@ -0,0 +1,102 @@
+import sublime_plugin
+
+from .core.configurations import is_supported_syntax
+from .core.protocol import Request, Range, DocumentHighlightKind
+from .core.clients import client_for_view
+from .core.documents import get_document_position
+from .core.settings import settings
+
+import sublime  # only for typing
+try:
+    from typing import List, Dict
+    assert List and Dict
+except ImportError:
+    pass
+
+
+_kind2name = {
+    DocumentHighlightKind.Unknown: "unknown",
+    DocumentHighlightKind.Text: "text",
+    DocumentHighlightKind.Read: "read",
+    DocumentHighlightKind.Write: "write"
+}
+
+
+class DocumentHighlightListener(sublime_plugin.ViewEventListener):
+
+    @classmethod
+    def is_applicable(cls, settings):
+        syntax = settings.get('syntax')
+        return syntax and is_supported_syntax(syntax)
+
+    def __init__(self, view: sublime.View) -> None:
+        super().__init__(view)
+        self._initialized = False
+        self._enabled = False
+        self._version = 0
+
+    def on_selection_modified_async(self) -> None:
+        if not self._initialized:
+            self._initialize()
+        if self._enabled:
+            self._clear_regions()
+            self._queue()
+
+    def _queue(self) -> None:
+        self._version += 1
```

Comment:
I'm sure it would take a lot of editing to hit the max bounds of `_version`, but I wonder if storing and checking the position of the cursor(s) could work equally well?